src/main/java/io/jenkins/plugins/opentelemetry/job/workflow/steps/WithSpanAttribute.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright The Original Author or Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.jenkins.plugins.opentelemetry.job.workflow.steps;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import hudson.Extension;
+import hudson.ExtensionList;
+import hudson.model.Item;
+import hudson.model.ItemGroup;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.util.ListBoxModel;
+import io.jenkins.plugins.opentelemetry.OtelUtils;
+import io.jenkins.plugins.opentelemetry.job.OtelTraceService;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.AttributeType;
+import io.opentelemetry.api.trace.Span;
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+import org.jenkinsci.plugins.workflow.steps.*;
+import org.kohsuke.stapler.AncestorInPath;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+import javax.annotation.Nonnull;
+import java.util.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+@Extension
+public class WithSpanAttribute extends Step {
+    private final static Logger logger = Logger.getLogger(WithSpanAttribute.class.getName());
+
+    String key;
+    Object value;
+    AttributeType type;
+
+    @DataBoundConstructor
+    public WithSpanAttribute() {
+
+    }
+
+    @Override
+    public StepExecution start(StepContext context) throws Exception {
+        AttributeType type = this.type;
+        if (type == null) {
+            if (value instanceof Boolean) {
+                type = AttributeType.BOOLEAN;
+            } else if (value instanceof Double) {
+                type = AttributeType.BOOLEAN;
+            } else if (value instanceof Long || value instanceof Integer) {
+                type = AttributeType.LONG;
+            } else {
+                type = AttributeType.STRING;
+            }
+        }
```

Comment:
What about a factory and specialized implementation for each type? this will avoid boxing/unboxing and make code simpler. the use of converters is another option.

Reply:
Can you make a proposal? I thought about a converter API similar to what can be found in XStream but it felt heavyweight to me.